csrc/host_ir/executor.cpp

```diff
@@ -202,6 +202,31 @@ void HostIrExecutor::handle(ForLoop* for_loop) {
   }
 }
 
+namespace {
+
+void handleWithExpressionEvaluator(
+    Expr* expr,
+    ExpressionEvaluator& expr_evaluator) {
+  for (auto input : ir_utils::filterByType<TensorView>(expr->inputs())) {
+    NVF_ERROR(
+        expr_evaluator.isKnown(input),
+        "input ",
+        input->toString(),
+        " of the expression ",
+        expr->toString(),
+        "must be precomputed before being retrieved");
+  }
+  for (auto output : expr->outputs()) {
+    expr_evaluator.bind(output, expr_evaluator.evaluate(output), true);
+  }
+}
+
+} // namespace
+
+void HostIrExecutor::handle(SliceOp* slice_op) {
+  return handleWithExpressionEvaluator(slice_op, expr_evaluator_);
```

Comment:
this indirection is motivated by the fact that other ops, e.g. matmul, will use the same auxiliary function `handleWithExpressionEvaluator`
See https://github.com/NVIDIA/Fuser/pull/2546